updateStableIds/src/main/java/org/reactome/release/update_stable_ids/UpdateStableIds.java

```diff
@@ -0,0 +1,163 @@
+
+public class UpdateStableIds {
+
+	private static final Logger logger = LogManager.getLogger();
+	
+	@SuppressWarnings("unchecked")
+	public static void stableIdUpdater(MySQLAdaptor dbaSlice, MySQLAdaptor dbaPrevSlice, MySQLAdaptor dbaGkCentral, Long personId) throws Exception {
+		
+		logger.info("Generating InstanceEdits for " + dbaSlice.getDBName() + " and " + dbaGkCentral.getDBName());
+		// Instance Edits for test_slice and gk_central
+		String creatorName = "org.reactome.release.update_stable_ids";
+		GKInstance sliceIE = InstanceEditUtils.createInstanceEdit(dbaSlice, personId, creatorName);
+		GKInstance gkCentralIE = InstanceEditUtils.createInstanceEdit(dbaGkCentral, personId, creatorName);
+
+		// At time of writing (December 2018), test_slice is a non-transactional database. This check has been put in place as a safety net in case that changes.
+		if (dbaSlice.supportsTransactions()) {
+			dbaSlice.startTransaction();
+		}
+		dbaGkCentral.startTransaction();
+		
+		//TODO:Should legacy regulation instances be updated?
+		//TODO:Ensure that re-run handling is implemented (backing up DB)
+		
+		// These two Lists were originally used to determine what classes' instances should be updated.
+		// Since all 'accepted' classes turned out to encompass all PhysicalEntity and Event classes, and all
+		// 'excluded' classes were Regulation classes, the filter was not needed anymore.
+//		final List<String> acceptedClassesWithStableIdentifiers = Arrays.asList("Pathway","SimpleEntity","OtherEntity","DefinedSet","Complex","EntityWithAccessionedSequence","GenomeEncodedEntity","Reaction","BlackBoxEvent","CandidateSet","Polymer","Depolymerisation","Polymerisation","Drug","FailedReaction","EntitySet");
+//		List<String> excludedClassesWithStableIdentifiers = Arrays.asList("PositiveGeneExpressionRegulation","PositiveRegulation","NegativeRegulation","Requirement","NegativeGeneExpressionRegulation");
+//		ResultSet classesWithStableIdentifiers = dbaGkCentral.executeQuery("SELECT DISTINCT _class FROM DatabaseObject WHERE StableIdentifier IS NOT NULL", null);		
+
+		logger.info("Fetching all Event and PhysicalEntity instances");
+		// Get all Event and PhysicalEntity instances and combine them into one large List
+		Collection<GKInstance> eventInstances = dbaSlice.fetchInstancesByClass(Event);
+		Collection<GKInstance> physicalEntityInstances = dbaSlice.fetchInstancesByClass(PhysicalEntity);
+		List<GKInstance> sliceInstances = new ArrayList<GKInstance>();
+		sliceInstances.addAll(eventInstances);
+		sliceInstances.addAll(physicalEntityInstances);
+
+		int incrementCount = 0;
+		logger.info("Total instances to check: " + sliceInstances.size());
+		for (GKInstance sliceInstance : sliceInstances) {
+			logger.info("Checking " + sliceInstance);
+			GKInstance gkCentralInstance = (GKInstance) dbaGkCentral.fetchInstance(sliceInstance.getDBID());
+			GKInstance prevSliceInstance = (GKInstance) dbaPrevSlice.fetchInstance(sliceInstance.getDBID());
```

Comment:
The fetchInstance method looks like it returns a GKInstance object, so no need to cast for these two lines.